com.unity.render-pipelines.high-definition/Runtime/Material/Water/Water.hlsl

```diff
@@ -127,15 +113,141 @@ BSDFData ConvertSurfaceDataToBSDFData(uint2 positionSS, SurfaceData surfaceData)
     bsdfData.perceptualRoughness = PerceptualSmoothnessToPerceptualRoughness(surfaceData.perceptualSmoothness);
     bsdfData.roughness = PerceptualRoughnessToRoughness(bsdfData.perceptualRoughness);
 
-    bsdfData.foamColor = surfaceData.foamColor;
+    bsdfData.foam = surfaceData.foam;
     bsdfData.refractionColor = surfaceData.refractionColor;
     bsdfData.tipThickness = surfaceData.tipThickness;
 
-    bsdfData.specularSelfOcclusion = surfaceData.specularSelfOcclusion;
-
     return bsdfData;
 }
 
+//-----------------------------------------------------------------------------
+// conversion function for deferred
+
+// Layout of the water gbuffer
+// GBuffer0 R16G16B16A16 (64 bits)
+// - GBuffer0: Bits 64 -> 32: Diffuse color (11 11 10)
+// - GBuffer0: Bits 31 -> 16:  Empty
+// - GBuffer0: Bits 15 -> 0: foam value
+
+// GBuffer1 R16G16B16A16 (64 bits)
+// - GBuffer1: Bits 63 -> 32: Complete Normal Surface Gradients
+// - GBuffer1: Bits 31 -> 0: Low frequency Normal Surface Gradients
+
+// GBuffer2 R32G32
```

Comment:
32/32 is really slow on multiple platform

Reply:
It is R16G16B16A16, old comment